the-collab-lab/tcl-17-smart-shopping-list
# Bl change firestore

## Description

This PR will do the following three things:
1. Update the firestore config keys
2. Show the user their 3 word token, so they can share it and invite other users to join their list
3. Add meta data to allow open graph protocol functionality - eg: sharing the application url will also share an image preview of the application



## Related Issue
closes [#83](https://github.com/the-collab-lab/tcl-17-smart-shopping-list/issues/83)



## Acceptance Criteria
- [ ] The user's current token is displayed above their list
- [ ] The user sees a message instructing them on the purpose of the three word token 
- [ ] The user can select an icon that enables `copy-to-clipboard` for their token

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|  ✓ | :sparkles: New feature     |

## Updates

### Before
<img width="310" alt="before-adding-token" src="https://user-images.githubusercontent.com/36529777/111057113-af101080-8439-11eb-93a1-d3ad028aaa6b.PNG">

### After
<img width="336" alt="after-adding-token" src="https://user-images.githubusercontent.com/36529777/111057723-82aac300-843e-11eb-920e-5e167b44223d.PNG">

## Testing Steps / QA Criteria
* One way to test the site is to click on the `details` button below to open the `Netlify preview`
![Screen Shot 2021-01-12 at 6 05 51 PM](https://user-images.githubusercontent.com/14856004/104389400-ff81f500-5500-11eb-95ae-4c0169a069cb.png)

* Otherwise from your terminal, pull down this branch with `git pull origin BL-change-firestore` and check that branch out with `git checkout BL-change-firestore`
* Run `npm start` in your terminal to launch the application in the browser 
1. Create a new list
2. Confirm that once you create a new list, the load screen is replaced by the list view - *(this will confirm that the new firestore keys are working as expected)*
3. Confirm that you see the message `Invite a friend to join your list by sharing this three word token: token token token` at the top of your list view - *(the three word token displayed at the end of the message will be different for each list)*
4. Click on the clipboard icon, open a text document of your choice (google docs, microsoft word, etc.) and paste *(right click and paste, ctrl+v or cmd+v)* to confirm that your three word token was successfully copied to your clipboard
